lib/components/ImportWallet.js

```diff
@@ -1,19 +1,80 @@
 import React, { PureComponent } from 'react';
-import { Text, Input, Button } from '@bpanel/bpanel-ui';
+import { Dropdown, Header, Text, Input, Button } from '@bpanel/bpanel-ui';
+import TerminalView from './TerminalView';
+import { importWallet as style } from '../style';
 
 class ImportWallet extends PureComponent {
   constructor(props) {
     super(props);
+    this.state = {
+      walletName: '',
+      selectedHardware: '',
+      derivationPath: '',
+      advanced: false,
+    };
+
+    // TODO: add trezor
+    this.dropdownOptions = ['ledger'];
+  }
+
+  update(key, value) {
+    const { state } = this;
+    this.setState(
+      Object.assign({}, state, {
```

Comment:
interesting. Why is the `Object.assign` necessary? I don't actually know for sure, but I would've thought `this.setState` takes care of this for you. Was mutability causing issues elsewhere?

Reply:
Removed the extra object allocation